eventlogger/src/main/java/gov/cms/ab2d/eventlogger/utils/UtilMethods.java

```diff
@@ -43,4 +48,43 @@ public static String hashIt(String val) {
     public static String hashIt(InputStream stream) throws IOException {
         return Hex.encodeHexString(DigestUtils.sha256(stream));
     }
+
+    public static <T> boolean allEmpty(List<T>...events) {
+        for (List<T> e : events) {
+            if (e != null && !e.isEmpty()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static String getURL(HttpServletRequest req) {
+        String method = req.getMethod();             // GET, POST
+        String scheme = req.getScheme();             // http
+        String serverName = req.getServerName();     // hostname.com
+        int serverPort = req.getServerPort();        // 80
+        String contextPath = req.getContextPath();   // /mywebapp
+        String servletPath = req.getServletPath();   // /servlet/MyServlet
+        String pathInfo = req.getPathInfo();         // /a/b;c=123
+        String queryString = req.getQueryString();   // d=789
```

Comment:
The API is behind a load balancer, so we'll need to take the `X-Forwarded` headers into account: https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/x-forwarded-headers.html